src/restconfig/src/main/java/org/geoserver/rest/catalog/AvailableResourcesConverter.java

```diff
@@ -90,14 +100,34 @@ private void writeJSON(AvailableResources t, HttpOutputMessage outputMessage)
 
     protected void writeXML(AvailableResources t, HttpOutputMessage outputMessage)
             throws IOException {
-        Element root = new Element("list");
-        final Document doc = new Document(root);
-        XMLOutputter outputter = new XMLOutputter(Format.getPrettyFormat());
-
-        t.stream().map(name -> new Element(t.getName()).addContent(name)).forEach(root::addContent);
 
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        try {
+            XMLOutputFactory factory = XMLOutputFactory.newFactory();
+            XMLStreamWriter writer = factory.createXMLStreamWriter(out, "UTF-8");
+            writer.writeStartDocument();
+            writer.writeStartElement("list");
+            for (String name : t) {
+                writer.writeStartElement(t.getName());
+                writer.writeCharacters(name);
+                writer.writeEndElement();
+            }
+            writer.writeEndElement();
+            writer.writeEndDocument();
+        } catch (XMLStreamException | FactoryConfigurationError e) {
+            throw new IOException(e);
+        }
+        // indent and write. If we didn't indent, the above could just write to
+        // outputMessage.getBody()
+        Transformer transformer;
         try (OutputStream os = outputMessage.getBody()) {
-            outputter.output(doc, os);
+            transformer = TransformerFactory.newInstance().newTransformer();
+            transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            // transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "4");
```

Comment:
Needed?

Reply:
AFAIK that's how you serialize a DOM to XML? if you mean the `OutputKeys.INDENT` property then yes, in order to preserve the behavior of serializing with indentation.

Reply:
I meant the commented source code.

Reply:
Ah no, that's not needed :) removed and updated the pr.